ivy_tests/test_ivy/helpers/function_testing.py

```diff
@@ -678,22 +680,62 @@ def test_frontend_function(
     args_np, kwargs_np = kwargs_to_args_n_kwargs(
         num_positional_args=test_flags.num_positional_args, kwargs=all_as_kwargs_np
     )
-
-    # extract all arrays from the arguments and keyword arguments
-    arg_np_vals, args_idxs, c_arg_vals = _get_nested_np_arrays(args_np)
-    kwarg_np_vals, kwargs_idxs, c_kwarg_vals = _get_nested_np_arrays(kwargs_np)
-    # make all lists equal in length
-    num_arrays = c_arg_vals + c_kwarg_vals
-    if len(input_dtypes) < num_arrays:
-        input_dtypes = [input_dtypes[0] for _ in range(num_arrays)]
-    if len(test_flags.as_variable) < num_arrays:
-        test_flags.as_variable = [test_flags.as_variable[0] for _ in range(num_arrays)]
-    if len(test_flags.native_arrays) < num_arrays:
-        test_flags.native_arrays = [
-            test_flags.native_arrays[0] for _ in range(num_arrays)
-        ]
+    # list and dict
+    # have to do type cast stuff here
 
     with BackendHandler.update_backend(backend_to_test) as ivy_backend:
+        target_args_np = copy.deepcopy(args_np)
+        target_kwargs_np = copy.deepcopy(kwargs_np)
+        backend_supported_dtypes = (
+            t_globals.CURRENT_RUNNING_TEST.supported_device_dtypes[backend_to_test][
+                on_device
+            ]
+        )
+        if input_dtypes[0] not in backend_supported_dtypes["valid"]:
```

Comment:
I don't think we should be having a block for this as discussed, it's much better to just set the casting modes and the input would be casted to a necessary dtype